src/java.base/share/classes/java/util/zip/DeflaterInputStream.java

```diff
@@ -70,6 +83,9 @@ private void ensureOpen() throws IOException {
     /**
      * Creates a new input stream with a default compressor and buffer
      * size.
+     * <p>
+     * The default compressor will be closed when this input stream
+     * is {@linkplain #close() closed}.
```

Comment:
There is another case where "the default compressor" may be mis-read to mean a system-wide compressor.